src/Uns.ts

```diff
@@ -270,33 +272,22 @@ export default class Uns extends NamingService {
   }
 
   async getTokenUri(tokenId: string): Promise<string> {
-    const [resultOrErrorL1, resultOrErrorL2] = await Promise.all([
-      this.unsl1.getTokenUri(tokenId).catch((err) => err),
-      this.unsl2.getTokenUri(tokenId).catch((err) => err),
-    ]);
-    if (resultOrErrorL2 instanceof Error) {
-      if (
-        !resultOrErrorL2.message.includes(
-          'ERC721Metadata: URI query for nonexistent token',
-        )
-      ) {
-        throw resultOrErrorL2;
+    // Kick off both requests concurrently and wrap the results to avoid unhandled promise rejections.
+    for await (const result of [
+      wrapResult(() => this.unsl1.getTokenUri(tokenId)),
+      wrapResult(() => this.unsl2.getTokenUri(tokenId)),
+    ]) {
+      const tokenUri = unwrapResult(result);
+      if (tokenUri) {
+        return tokenUri;
```

Comment:
Order is important, if there is token on L2, it should be default, if not, than L1

Reply:
oh, good catch, I remembered about the order but forgot to apply it in my final version